vortex-datafusion/src/convert/ranges.rs

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::ops::Range;
```

Comment:
this just got moved out of opener.rs since it was getting long, and having these tests there was distracting